src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -190,12 +190,28 @@ impl KvmVcpu {
             Some(kvi) => kvi,
             None => Self::default_kvi(vm_fd, self.index)?,
         };
+        self.kvi = state.kvi;
 
         self.init_vcpu(&kvi)?;
+
+        // If KVM_REG_ARM64_SVE_VLS is present it needs to
+        // be set before vcpu is finalized.
+        if let Some(sve_vls_reg) = state
+            .regs
+            .iter()
+            .find(|reg| reg.id == KVM_REG_ARM64_SVE_VLS)
+        {
+            set_register(&self.fd, sve_vls_reg).map_err(KvmVcpuError::RestoreState)?;
+        }
+
         self.finilize_vcpu(&kvi)?;
-        self.kvi = state.kvi;
 
-        for reg in state.regs.iter() {
+        // KVM_REG_ARM64_SVE_VLS needs to be skipped after vcpu is finalized.
```

Comment:
maybe mention that it is skipped because we already handled it above

Reply:
It is skipped not because it is already handled, but because it is not valid to set it after `finalize` is called.
Added a comment about handling it above.